src/Raven.Server/Monitoring/OpenTelemetry/DatabaseWideMetrics.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Exceptions.Database;
+using Raven.Server.Config.Categories;
+using Raven.Server.Documents;
+using Raven.Server.Monitoring.Snmp.Objects.Database;
+
+namespace Raven.Server.Monitoring.OpenTelemetry;
+
+public class DatabaseWideMetrics : MetricsBase
+{
+    public const int IgnoreIndex = 0;
+    private readonly DatabasesLandlord _databaseLandlord;
+    private readonly string _databaseName;
+    private readonly SemaphoreSlim _locker = new(1, 1);
+    private bool _attached;
+    private readonly HashSet<string> _loadedIndexes = new(StringComparer.OrdinalIgnoreCase);
+
+    //Meter instances are universally shared across all indexes, as identification of indexes or databases occurs through measurement tags.
+    private static readonly Lazy<Meter> IndexesMeter = new(() => new Meter(Constants.IndexMeter, "1.0.0"));
```

Comment:
what is this 1.0.0 here?

Reply:
OpenTelemetry gives you possibility to version meter implementation.
https://opentelemetry.io/docs/specs/otel/versioning-and-stability/